Engine/src/objects/rule/Rule.java

```diff
@@ -4,13 +4,14 @@
 import properties.activition.Activation;
 
 import java.util.Map;
+import java.util.Set;
 
 public class Rule {
     private final String name;
     private final Activation activation;
-    private final Map<String,Action> actions;
+    private final Set<Action> actions;
```

Comment:
I didn't change that because our toString overrides based on how the UI wants to prints this information. So, for this exercise they are build to compose the info for a console app, and for the next exercise we will need to change them. According to Aviad's last lesson, thats considered a bad practice.
So I think in general we need to change every toString override in the project and adjust them to the DOT object creation.